dataprofiler/data_readers/json_data.py

```diff
@@ -268,36 +276,36 @@ def _load_data_from_file(self, input_file_path):
                 )
             return data
 
-    def _get_data_as_records(self, data):
+    def _get_data_as_records(self, data: List) -> List[str]:
         """
         Extract the data as a record format.
 
         :param data: raw data
         :type data: list
         :return: dataframe in record format
         """
-        data = self._get_data_as_df(data)
-        data = data.to_dict(orient="records", into=OrderedDict)
-        for i, sample in enumerate(data):
-            data[i] = json.dumps(
+        data_df: pd.DataFrame = self._get_data_as_df(data)
+        data_records: List = data_df.to_dict(orient="records", into=OrderedDict)
+        for i, sample in enumerate(data_records):
+            data_records[i] = json.dumps(
```

Comment:
no longer overwriting variable names... was mypy complaining about this and that is the reason for the change?

Reply:
The original code kept reassigning the types of data stored in `data` but mypy doesn't let you do that (each variable can only have 1 type in its lifetime). We could make `data` be typed as a union of all the types it takes on but I was thinking that would be confusing.

Reply:
The concern would be now we have the data 3x in memory at one time. We could use a defined type which is the union. This might be a decent case for casting or allowing data to be Any

Reply:
>  data 3x in memory at one time
Yeah, this is what the code was trying to avoid as it was written last time. 

Couple options in order of how robust they are:
- `Union[]` is an option
- casting
- `Any`

Reply:
+1